File: tools/templates/integration/src/main/scala/test1/modules.scala
package _PROJ_CLASS_.test1

import org.tresamigos.smv._

object S1 extends SmvModule("") with SmvOutput {
  override def requiresDS = Seq(input.table)

  override def run(i: runParams) = {
    i(input.table)
  }
}

object S2 extends SmvModule("") with SmvOutput {
  override def requiresDS = Seq(S1)

  override def run(i: runParams) = {
    i(S1)
  }
}

File: tools/templates/test/src/main/python/test2/modules.py
from smv import *

from integration.test.test2 import input

class M1(SmvModule):
    def requiresDS(self):
        return [input.table]

    def run(self, i):
        return i[input.table]

class M2(SmvModule, SmvOutput):
    def requiresDS(self):
        return [M1]

    def run(self, i):
        return i[M1]
